pkg/resourcemanager/controller/managedresource/reconciler.go

```diff
@@ -466,6 +466,32 @@ func (r *Reconciler) updateConditionsForDeletion(ctx context.Context, mr *resour
 	return updateConditions(ctx, r.SourceClient, mr, conditionResourcesHealthy, conditionResourcesProgressing)
 }
 
+func preserveAnnotation(kind string, apiVersion string, current *unstructured.Unstructured, desired *unstructured.Unstructured, annotationKey string, path ...string) error {
+	if current.GetKind() != kind || current.GetAPIVersion() != apiVersion {
+		return nil
+	}
+	currentAnnotations, found, err := unstructured.NestedStringMap(current.Object, path...)
+	if err == nil && !found {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed getting %s from %s/%s %s : %w", strings.Join(path, "."), apiVersion, kind, current.GetName(), err)
+	}
+	value, ok := currentAnnotations[annotationKey]
+	if !ok {
+		return nil
+	}
+	desiredAnnotations, found, err := unstructured.NestedStringMap(desired.Object, path...)
+	if !found {
+		desiredAnnotations = map[string]string{}
+	}
+	if err != nil {
+		return fmt.Errorf("failed getting %s from %s/%s %s : %w", strings.Join(path, "."), apiVersion, kind, desired.GetName(), err)
+	}
+	err = unstructured.SetNestedField(desired.Object, mergeLabels(map[string]string{annotationKey: value}, desiredAnnotations), path...)
+	return fmt.Errorf("failed setting annotation %s at %s for %s/%s %s: %w", annotationKey, strings.Join(path, "."), apiVersion, kind, desired.GetName(), err)
```

Comment:
```suggestion
	if err := unstructured.SetNestedField(desired.Object, mergeLabels(map[string]string{annotationKey: value}, desiredAnnotations), path...); err != nil {
		return fmt.Errorf("failed preserving annotation %q at %q for %s/%s %s: %w", annotationKey, strings.Join(path, "."), apiVersion, kind, desired.GetName(), err)
	}
	return nil
```

Error check missing?